modules/swagger-codegen/src/main/resources/typescript-fetch/api.mustache

```diff
@@ -212,7 +212,12 @@ export const {{classname}}FetchParamCreator = function (configuration?: Configur
 
     {{/hasFormParams}}
     {{#bodyParam}}
+            {{^consumes}}
             localVarHeaderParameter['Content-Type'] = 'application/json';
+            {{/consumes}}
+            {{#consumes.0}}
+            localVarHeaderParameter['Content-Type'] = '{{mediaType}}';
```

Comment:
@masaeedu Please use {{{mediaType}}} to use the original value fo the media type as I remember there are edge cases causing issues with the HTML-escaped values.

Reply:
@wing328 Good catch. Should we be doing this for things like `{{className}}`, `{{paramName}}` etc. as well?

Reply:
@masaeedu yes please

Reply:
FYI. I've filed https://github.com/swagger-api/swagger-codegen/pull/7073 to clean up {{mediaType}} in other templates.